src/Documents/ListRecordsDocument.php

```diff
@@ -10,14 +11,23 @@
 class ListRecordsDocument extends OaiDocument
 {
 
-    private OaiRecordFormatter $formatter;
+    private ?OaiRecordFormatter $formatter = null;
 
-    public function __construct(OaiRecordFormatter $formatter)
+    public function __construct(?OaiRecordFormatter $formatter = null)
```

Comment:
The refactor here is so that I can instantiate a `ListRecordsDocument` before I know which record formatter I'm going to use. I have done this because I would like to be able to add errors to the document, and I do this before knowing what the `metadataPrefix` is.

Reply:
Sweet, thanks for the note!